docs/static/pipeline-viewer.asciidoc

```diff
@@ -0,0 +1,107 @@
+[role="xpack"]
+[[logstash-pipeline-viewer]]
+=== Pipeline Viewer UI
+
+NOTE: The pipeline viewer is an {xpack} feature that requires a
+paid {xpack} license. See the
+https://www.elastic.co/subscriptions[Elastic Subscriptions] page for
+information about obtaining a license.
+
+//REVIEWERS: Not sure what license is required here. Does this come with basic?
+
+The pipeline viewer in {xpack} provides a simple way for you to visualize and
+monitor the behavior of complex Logstash pipeline configurations. Within the
+pipeline viewer, you can explore a directed acyclic graph (DAG) representation
+of the overall pipeline topology, data flow, and branching logic. The diagram
+is overlayed with important metrics, like events per second and time spent in
+milliseconds, for each plugin in the view.
+
+The diagram includes visual indicators to draw your attention to potential
+bottlenecks in the pipeline, making it easy for you to diagnose and fix
+problems.
+
+Before using the pipeline viewer, you need to
+{logstash-ref}/setup-xpack.html[set up {xpack}] and configure
+{xpack-ref}/monitoring-logstash.html[Logstash monitoring]. 
+
+//REVIEWERS: Is anything else required to use the pipeline viewer?
+
+[float]
+==== View the pipeline diagram
+
+To view the pipeline diagram:
+
+. In Logstash, start the Logstash pipeline that you want to monitor.
++
+Assuming that you've set up Logstash monitoring, Logstash will begin shipping
+metrics to the monitoring cluster.
+
+. Navigate to the Monitoring tab in Kibana.
++
+You should see a Logstash section. 
++
+image::static/docs/images/monitoring-ui.png[Monitoring UI]
+
+. Click the *Pipelines* link under Logstash to see all the pipelines that are
+being monitored.
++
+Each pipeline is identified by a pipeline ID (`main` by default). For each
+pipeline, you'll see a list of all versions of the pipeline running during
+the specified time range.
++
+image::static/docs/images/pipeline-viewer-overview.png[Pipeline Overview]
++
+The version information is auto-generated by Logstash. Each time you modify a
+pipeline, Logstash generates a new version hash. Viewing different versions
```

Comment:
Lets also mention that we don't store multiple versions of the pipeline configuration itself, just the pipeline stats from latest one.